monitor.cpp

```diff
@@ -3,36 +3,55 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <string>
+#include <vector>
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
+void CheckVitals::displayWarning(const std::string& message) {
+    std::cout << message << "\n";
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        std::cout << "\r* " << std::flush;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        std::cout << "\r *" << std::flush;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
     }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+// Helper functions to check each vital
+bool isTemperatureCritical(double temperature) {
+    return temperature > 102 || temperature < 95;
+}
+
+bool isPulseRateOutOfRange(double pulseRate) {
+    return pulseRate < 60 || pulseRate > 100;
```

Comment:
Though the tool doesn't say it, the comparison logic here is repeated from the temperature check.
E.g., if `<` changes to `<=`, you will need to change in multiple places. Try isolating the range comparison logic into its own function